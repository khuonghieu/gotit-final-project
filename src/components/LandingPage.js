import React from 'react';
import { connect } from 'react-redux';
import { Header, Button } from '@gotitinc/design-system';
import UserProfile from './User/UserProfile';
import { signInModal, signUpModal } from '../actions/changeModal';

function mapStateToProps(state) {
  return {
    user: state.user,
  };
}

const mapDispatchToProps = (dispatch) => ({
  signUpModal: () => dispatch(signUpModal),
  signInModal: () => dispatch(signInModal),
});

export function LandingPage({ user, signUpModal, signInModal }) {
  return (
    <div>
      <Header fullWidth>
        <Header.Main>
          <Header.Left>
            <UserProfile user={user} />
          </Header.Left>
          <Header.Right>
            <Button onClick={signUpModal}>Sign up modal</Button>
            <Button onClick={signInModal}>Sign in modal</Button>
          </Header.Right>
        </Header.Main>
      </Header>
    </div>
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(LandingPage);
