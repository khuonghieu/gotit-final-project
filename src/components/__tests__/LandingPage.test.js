import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { LandingPage } from '../LandingPage';

configure({ adapter: new Adapter() });

describe('components/LandingPage.js', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      user: {
        loggedIn: false,
        currentUser: null,
      },
      signUpModal: jest.fn(),
      signInModal: jest.fn(),
    };
  });

  const setup = () => {
    wrapper = shallow(
      <LandingPage {...props} />,
    );
  };

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
});
