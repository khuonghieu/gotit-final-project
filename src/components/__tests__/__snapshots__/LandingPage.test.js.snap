// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LandingPage.js should render correctly 1`] = `ShallowWrapper {}`;
